server/etcdserver/server.go

```diff
@@ -1163,11 +1163,9 @@ func (s *EtcdServer) NewUberApplier() apply.UberApplier {
 }
 
 func verifySnapshotIndex(snapshot raftpb.Snapshot, cindex uint64) {
-	verify.Verify(func() {
-		if cindex != snapshot.Metadata.Index {
-			panic(fmt.Sprintf("consistent_index(%d) isn't equal to snapshot index (%d)", cindex, snapshot.Metadata.Index))
-		}
-	})
+	verify.VerifyPrototype(func() bool {
+		return cindex != snapshot.Metadata.Index
+	}, fmt.Sprintf("consistent_index(%d) isn't equal to snapshot index (%d)", cindex, snapshot.Metadata.Index), nil)
```

Comment:
We don't want to render `fmt.Sprintf` not construct `details map[string]any` if `IsVerificationEnabled` is false.

Reply:
How about this newly pushed one?  By returning them in the func, the string should not be rendered and the map won't be constructed prematurely.